src/openttd.cpp

```diff
@@ -1092,6 +1092,18 @@ void SwitchToMode(SwitchMode new_mode)
 			break;
 
 		case SM_RESTARTGAME: // Restart --> 'Random game' with current settings
+			if (_game_mode == GM_EDITOR && _file_to_saveload.abstract_ftype == FT_SAVEGAME || _file_to_saveload.abstract_ftype == FT_SCENARIO) {
+				_game_mode == GM_EDITOR ? _switch_mode = SM_LOAD_SCENARIO : _switch_mode = SM_LOAD_GAME;
```

Comment:
```suggestion
				_switch_mode = _game_mode == GM_EDITOR ? SM_LOAD_SCENARIO : SM_LOAD_GAME;
```
and similarly for below

Reply:
Done